text/2872-github-access-policy.md

```diff
@@ -0,0 +1,82 @@
+- Feature Name: rust-lang_github_org_access_policy 
+- Start Date: 2020-03-02 
+
+# Summary
+[summary]: #summary
+
+This RFC proposes a policy for managing permissions to the [Rust-Lang GitHub Organization](https://www.github.com/rust-lang) and repositories within this organization.
+
+This RFC was written in consultation with the Governance Working Group and the Infrastructure team. Most discussion took place on [this issue](https://github.com/rust-lang/wg-governance/issues/4) and [this pull request](https://github.com/rust-lang/wg-governance/pull/42).
+
+# Motivation
+[motivation]: #motivation
+
+Access control for the [Rust-Lang GitHub Organization](https://www.github.com/rust-lang) and repositories within that organization is currently managed either through the [rust-lang team database][db], or ad-hoc via the GitHub UI by the org administrators. We need a policy that defines how these accesses are granted and managed. This will allow us to have greater security in permissions to our GitHub org, and provide transparency and clarity on how access is managed.
+
+[db]: https://github.com/rust-lang/team/
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+## Rust-Lang GitHub Permissions Policy
+
+This policy applies to both the [Rust-Lang GitHub Organization](https://github.com/rust-lang/) and all repositories within that organization.
+
+### Rust-Lang Organization
+
+Access to the Rust-Lang GitHub organization is managed with the [rust-lang team database][db]. The team database is managed by the [team-repo-admins], whose policies are specified in the [Team Maintenance] documentation.
+
+Selected members of the [Infrastructure Team] can also be organization owners if their work requires it.
+
+All GitHub accounts used to interact with the Rust-Lang GitHub organization (owner or non-owner) must have 2FA enabled.
+
+[team-repo-admins]: https://github.com/rust-lang/team/blob/master/teams/team-repo-admins.toml
+[Team Maintenance]: https://forge.rust-lang.org/infra/team-maintenance.html
+[Infrastructure Team]: https://github.com/rust-lang/team/blob/master/teams/infra.toml
+
+### Rust-Lang Repositories
+
+Access to and permissions for repositories within the Rust-Lang organization must be administered through the [rust-lang team database][db]. Permissions should not be given to individuals, only to teams or groups.
```

Comment:
I suppose I don't have specific circumstances in mind. But, there are currently a number of repos with individual access and I'm sure there may be specific reasons that is the case.

Reply:
This is a key point of this RFC to stop giving individuals access. I have reviewed the existing individual permissions, and I don't believe they are all necessary (and to be clear I am one of those with individual access). There are different reasons, but generally I think they can be resolved by adding new teams or other arrangements.

Reply:
What "other arrangements" would you suggest?

A concrete way to solve this is to remove all individual permissions *now*, before the RFC is merged.

Reply:
> What "other arrangements" would you suggest?

There are options like using bot tooling, adding people to existing teams, adding already existing teams if the person is already a part of them, or removing people if they are no longer active.

> A concrete way to solve this is to remove all individual permissions now, before the RFC is merged.

There is already work underway to remove them, but I would prefer to not delay this RFC for that work to be finished. It might take many months to take care of every last entry. I would like to approve this to make it clear that no *new* individual permissions should be given.

Reply:
I still think that the wording here should be relaxed a bit (maybe "Other than in exceptional circumstances, ...") but I don't feel strong enough to block this.